packages/babel-traverse/src/scope/index.js

```diff
@@ -864,6 +868,14 @@ export default class Scope {
   }) {
     let path = this.path;
 
+    if (path.isPattern()) {
+      // We are inside some function parameters
+      throw path.buildCodeFrameError(
+        "[Internal error] It's not possible to inject a variable in function parameters. " +
+          "Please check in the stack trace of this error which plugin is causing it.",
```

Comment:
This might cause new errors, but it's better to explicitly fail than to generate invalid code.

For example, it will throw when compiling `function f(a = x()?.b) {}` with a new version of `@babel/traverse` and an old version of the optional chaining plugin.